<?xml version="1.0"?>
<launch>
  <param name="robot_description"
    command="$(find xacro)/xacro.py '$(find youbot_description)/robots/youbot_arm.urdf.xacro'" />
  
  <!-- Launch the inverse kinematic solver service -->
  <node name="ik_solver_service" pkg="ik_solver_service" type="ik_solver_service" output="screen" />

  <!-- send fake joint values -->
  <node name="joint_state_publisher" pkg="youbot_kinematics_rviz" type="ik_arm_state_publish" output="screen"/>

  <!-- reconfigure -->
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>

  <!-- Combine joint values -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher"/>

  <!-- Show in Rviz   -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find youbot_description)/launch/youbot.rviz" />

</launch>
